packages/headless/src/features/product-listing/product-listing-actions.ts

```diff
@@ -119,6 +143,54 @@ export const fetchProductListing = createAsyncThunk<
   }
 );
 
+export type StateNeededByFetchProductListingV2 = ConfigurationSection &
+  ProductListingV2Section &
+  Partial<
+    PaginationSection &
+      StructuredSortSection &
+      FacetSection &
+      NumericFacetSection &
+      CategoryFacetSection &
+      DateFacetSection &
+      FacetOptionsSection &
+      FacetOrderSection &
+      ContextSection
+  >;
+
+export interface FetchProductListingV2ThunkReturn {
+  /** The successful search response. */
+  response: ProductListingV2SuccessResponse;
+  analyticsAction: PreparableAnalyticsAction<
+    {analyticsType: AnalyticsType.Search},
+    StateNeededByFetchProductListingV2
+  >;
+}
+
+export const fetchProductListingV2 = createAsyncThunk<
+  FetchProductListingV2ThunkReturn,
+  void,
+  AsyncThunkProductListingV2Options<StateNeededByFetchProductListingV2>
+>(
+  'productlistingV2/fetch',
```

Comment:
Why do we provide multiple actions for search (execute, fetchPage, fetchMoreResults, fetchFacetValues...) and we want to get by with just 'fetch'?

Reply:
For PLP v2, only `productlistingV2/fetch` is used as an action for fetch product listings. This is the action reducers will handle.

The other ones you mention are used in the context of search.